File: 프로그래머스/3/42884. 단속카메라/단속카메라.js
function solution(routes) {
    var answer = 0;
    routes.sort((a,b)=>a[1]-b[1]);
    let curCam = -30001;
    for(let i=0; i<routes.length; i++){
        if(curCam<routes[i][0]){
            answer++;
            curCam=routes[i][1];
        }
    }
    return answer;
}

File: 프로그래머스/3/42884. 단속카메라/단속카메라.java
import java.util.*;
class Solution {
    public int solution(int[][] routes) {
        int answer = 0;
        Arrays.sort(routes,(a,b)->a[1]-b[1]);

        int out = -30001;
        for(int i=0; i<routes.length; i++){
            if(routes[i][0]>out){
                answer++;
                out = routes[i][1];
            }
        }
        return answer;
    }
}